components/Common/ThemeToggle/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { MoonIcon, SunIcon } from '@heroicons/react/24/outline';
+import { AccessibleIcon } from '@radix-ui/react-accessible-icon';
+import type { FC } from 'react';
+import { useIntl } from 'react-intl';
+
+import styles from './index.module.css';
+
+type ThemeToggleProps = {
+  toggleTheme: () => void;
```

Comment:
I think it is incorrect to call this toggleTheme. As this doesn't toggle the theme, hence why I said (and repeating myself again) to call it `onClick`.

Often I don't understand WHY when I ask you to do X, and you agree with X then you just do something completely different from asked?

Reply:
because it's not really onClick it's also onKeydown so I call props toggleTheme

Reply:
But it's not an `toggleTheme`, this Component does not toggle a theme.

You can just call it `onChange` or `inTeraction` or simply have two different handlers.

one called `onClick` and one called `onKeyDown` as props and simply expose them ;)

Reply:
> one called onClick and one called onKeyDown as props and simply expose them ;)

i prefer `onChange` it's less code but two props is more clear 👌🏻

Reply:
Right, but the thing is that hmm, nothing is really changing, right? And in theory we are "listening" to the user to interact with this Icon. Either by a keypress or a click (hence why I believe exposing two handlers is more appropriate).

Since you're being stubborn here (as usual), @bmuenzenmeyer what's your opinion in here? (if any)

Reply:
Sorry I didnt see this until now. My notifications were messed up somehow.

Let's look inward to help.  We are internally inconsistent about this. A non-exhaustive check of our current new components:

- [Select calls it `onChange`](https://github.com/nodejs/nodejs.org/blob/main/components/Common/Select/index.tsx#L33)
- [Notification calls it `onChange`](https://github.com/nodejs/nodejs.org/blob/main/components/Common/Notification/index.tsx#L10)
- [LanguageDropdown calls it `onClick`](https://github.com/nodejs/nodejs.org/blob/main/components/Common/LanguageDropDown/index.tsx#L16)

If it were me (one opinion), I'd probably call the function prop `onChange` or `handleChange`...but only wire it up to the `onClick` handler of the button. Button's being interactive elements, they should accept keyboard input automatically right? The tests could prove that. Also, `onkeydown` could spam events if someone holds it down right?

Reply:
>  Also, onkeydown could spam events if someone holds it down right?

Yes, so I can wrap `onChange` with an timeout

Reply:
As @bmuenzenmeyer said, we don't need an onKeyDown here, Augustin. accessible interactions are already handled by the `button` element.

Hence this should be called `onClick`.  Can you please just do as you're being asked?